backend/hitas/views/apartment_sale.py

```diff
@@ -87,14 +87,23 @@ def create(self, validated_data: dict[str, Any]) -> ApartmentSale:
                     Ownership.objects.select_related("apartment"),
                 ),
                 # Limit the fetched sales to only the first sale,
-                # as we only need that to figure out if the apartment is new
-                prefetch_first_sale("ownerships__apartment__"),
+                # as we only need that to figure out if the apartment is new.
+                # Ignore the sale we just created so that this apartment is treated as new.
+                prefetch_first_sale("ownerships__apartment__", ignore=[instance.id]),
```

Comment:
I'm having a really hard time understanding why we need to consider the apartment as new when it has been sold already

Reply:
It's because of this edge case:
1. Apartment has been completed AND has not been sold before -> it's still considered NEW
2. We made the first sale above AND **_it is in the past_** -> Rules now count the apartment as NOT NEW
3. If we would try to create conditions of sale for the new ownerships, this apartment that was just sold would not be identified as NEW, and no conditions of sale would be created for it.

Ideally `apartment.is_new` would be able to recognize apartments like this as new, but they would have to be separated from apartments that have only sold once but are not new anymore. I don't think we have a clear rule for this, unless I've missed something?

Reply:
Note that this only starts to matter once we remove `first_purchase_date` from apartment. Currently, I have `first_purchase_date` be set after the conditions of sale are created, which works out so that the apartment is new regardless of whether we ignore the previously created sale, since both the `first_purchase_date` and the first sale are checked in `apartment.is_new`.